### ISSUE TITLE
Does able to convert schematic file also?

### ISSUE THREAD

**Issue by mekesim** [urish/easyeda2kicad#2]
Thank you. It's quite a excellent tool. That can convert easyeda pcb file to kicad, include netlist.
Is there any way to covert PCB file to schematic file? Or directly convert schematic from easyeda?

**Comment by urish** [comment]
Thanks for your feedback! Schematic is not currently supported, only the board layout. If there is enough demand, I may add it in the future

**Comment by yangjuncode** [comment]
it will be awesome if both schema and pcb can export.

**Comment by mtiutiu** [comment]
Awesome indeed so I vote :+1:  for this one also.

**Comment by mtiutiu** [comment]
@urish Is there a plan for this one also? It would be helpful in case of big schematics as it gets tedious to be re-created by hand in KiCAD. Indeed PCB conversion is more important at this point but I think this one will be also in the future IMHO. 

I can help with testing of course but not on the development side 'cause I'm not a good programmer :smile:.

Thanks.

**Comment by urish** [comment]
@mtiutiu not sure yet - I'm checking with some friend who may want to look into it

**Comment by urish** [comment]
@karinchechik started working on this

**Comment by urish** [comment]
For the time being, the code only handles the conversions of wires in schematics. The command for converting the schematics is the same as PCB, and the program automatically detects the file type and performs the conversion accordingly:

`easyeda2kicad filename.json filename.sch`

Components, junctions, texts, etc. are not supported yet. Also multi-page schematics are not supported (only the first page will be converted).

**Comment by urish** [comment]
The latest release (1.6.0) also add supports for Junctions and No Connect flags in schematics.

**Comment by thecriticalpoint** [comment]
Thank you @urish. You made an incredibly useful thing.

Just out of curiosity is what would it take for you to be motivated enough to want to have it convert the easyEDA schematic? If that isn't a valid question in 2021 is there another way to perform this task with possibly another tool that didn't exist in Feb 2020. A lots's happened since then.

**Comment by urish** [comment]
@yaybee worked on it in #58, but unfortunately I haven't had the time yet to go over the changes and integrate them

**Comment by infosisio** [comment]
I would like to also express my interest for supporting schematic conversion.

**Comment by gsrobotics** [comment]
I would like to also express my interest in supporting schematic conversion.

**Comment by jrsphoto** [comment]
Yes, please. I know this is an old thread but if you could add schematic conversion you would save me weeks, of time..

**Comment by urish** [comment]
Unfortunately, I do not have the bandwidth to add this feature.

**Comment by yangjuncode** [comment]
here is a python version: https://github.com/uPesy/easyeda2kicad.py

**Comment by urish** [comment]
@yangjuncode thanks for sharing! Cool to see that your version also supports 3d module conversion.

**Comment by M10CUBE** [comment]
Guys this is getting better and better. I hope to see soon a final version that does it all. Congratulations to everybody.
Our team without KiCAD will never could do this contribute to the community: www.wizcube.eu
Easyeda is a place where useful material can be found and transferred to  KiCad to easy community designs.
Thanks again guys...

**Comment by boromyr** [comment]
Is there a hope that the schematic conversion will be implemented?

**Comment by KthProg** [comment]
Really need this. The only reason I'm even using KiCAD is because EasyEDA completely borked the PCB created from my schematic, so I need to restart with the schematic in better software.

**Comment by channelsixlabs** [comment]
One more vote for EasyEDA schematic conversion support. Thanks!

**Comment by nrodday** [comment]
+1 for the SCH conversion feature :)
